tree/dataframe/inc/ROOT/RDF/RDisplay.hxx

```diff
@@ -91,32 +91,38 @@ private:
    size_t fNMaxCollectionElements = 10; // threshold on number of elements in collections to be Print()
 
    ////////////////////////////////////////////////////////////////////////////
-   /// Appends a string if the T type is an arithmetic type.
+   /// Appends a cling::printValue call to the stringstream
+   /// This overload works for non-collection data types which are also not
+   /// trivially representable as strings.
    /// \tparam T the type of the event to convert
    /// \param[in] stream Where the conversion function call will be chained.
    /// \param[in] element The event to convert to its string representation
    /// \param[in] index To which column the event belongs to
    /// \return false, the event is not a collection
-   template <typename T, typename std::enable_if_t<std::is_floating_point<T>::value && !ROOT::Internal::RDF::IsDataContainer<T>::value, int> = 0>
+   template <typename T,
+             std::enable_if_t<!std::is_arithmetic_v<T> && !ROOT::Internal::RDF::IsDataContainer<T>::value, int> = 0>
    bool AddInterpreterString(std::stringstream &stream, T &element, const int &index)
    {
       stream << "*((std::string*)" << ROOT::Internal::RDF::PrettyPrintAddr(&(fRepresentations[index]))
-               << ") = std::to_string(*(" << fTypes[index] << "*)" << ROOT::Internal::RDF::PrettyPrintAddr(&element) << ");";
+             << ") = cling::printValue((" << fTypes[index] << "*)" << ROOT::Internal::RDF::PrettyPrintAddr(&element)
+             << ");";
       return false;
    }
 
    ////////////////////////////////////////////////////////////////////////////
-   /// Appends a cling::printValue call to the stringstream
+   /// Appends a string if the T type is an arithmetic type.
+   /// This overload works for arithmetic data types that are trivially
+   /// convertible to string.
    /// \tparam T the type of the event to convert
-   /// \param[in] stream Where the conversion function call will be chained.
    /// \param[in] element The event to convert to its string representation
    /// \param[in] index To which column the event belongs to
    /// \return false, the event is not a collection
-   template <typename T, typename std::enable_if_t<!std::is_floating_point<T>::value && !ROOT::Internal::RDF::IsDataContainer<T>::value, int> = 0>
-   bool AddInterpreterString(std::stringstream &stream, T &element, const int &index)
+   template <typename T, std::enable_if_t<std::is_arithmetic_v<T>, int> = 0>
```

Comment:
@vepadulano - you need to make sure that booleans are excluded

Reply:
Right! Apologies I had missed the previous comment about this issue, now is fixed in the latest commit